src/dawn/CodeGen/Cuda/ASTStencilBody.h

```diff
@@ -41,24 +41,18 @@ class ASTStencilBody : public ASTCodeGenCXX {
   const std::shared_ptr<iir::StencilInstantiation>& instantiation_;
   RangeToString offsetPrinter_;
   const std::unordered_map<int, Array3i>& fieldIndexMap_;
-  const iir::MultiStage& ms_;
+  const std::unique_ptr<iir::MultiStage>& ms_;
```

Comment:
some other function was requiring unque_ptr<>& . That is probably why is good idea to always stay in the smart ptr, because once we dereference it there is no way to call functionality that uses smart ptr &